tests/unit/test_tempest_extremes.py

```diff
@@ -577,6 +577,33 @@ def test_stitch_nodes_threshold_filters(
         assert result["stderr"] == "Mocked stderr output"
         assert result["returncode"] == 0
 
+    def test_stitch_nodes_values_from_dn(self) -> None:
+        """Check the values are being assigned properly from DetectNodesParameters."""
+        # Define the DetectNodes parameters to test against
+        output_commands = [
+            TEOutputCommand(var="v1", operator="min", dist=0.0),
+            TEOutputCommand(var="v2", operator="max", dist=3.0),
+        ]
+        dn_params = DetectNodesParameters(
+            output_file="file.txt", output_commands=output_commands
+        )
+
+        # Check defaults
+        tracker = TETracker()
+        assert tracker.stitch_nodes_parameters.in_file is None
+        assert tracker.stitch_nodes_parameters.in_fmt is None
+
+        # Check the auto-assignment works
+        tracker = TETracker(dn_params)
+        assert tracker.stitch_nodes_parameters.in_file == "file.txt"
+        assert tracker.stitch_nodes_parameters.in_fmt == ["lon", "lat", "v1", "v2"]
+
+        # Ensure defined values don't get overriden
+        sn_params = StitchNodesParameters(in_file="file2.txt", in_fmt=["a", "b"])
+        tracker = TETracker(dn_params, sn_params)
+        assert tracker.stitch_nodes_parameters.in_file == "file2.txt"
+        assert tracker.stitch_nodes_parameters.in_fmt == ["a", "b"]
+
```

Comment:
I think this is OK, but really we are testing 3 edge cases here so would prefer to see this as a test parameterised with `dn_params`, `sn_params`, and expected outputs.
However, given time constraints on the project I don't think this is a blocker for this PR.

Reply:
Thanks, would have been OK without, but definitely an improvement longer term, so thanks for adding this as well!